File: 프로그래머스/연습문제/Lv1/숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    cnts = [0 for i in range(10)]
    for Xi in X: cnts[int(Xi)] += 1

    answer = ""
    for Yi in Y:
        if cnts[int(Yi)] > 0:
            answer += Yi
            cnts[int(Yi)] -= 1

    answer = sorted(answer, reverse=True)

    if len(answer) == 0: return "-1"
    elif answer[0] == '0': return "0"
    else: return ''.join(answer)

File: 프로그래머스/연습문제/Lv1/숫자 짝꿍/숫자 짝꿍.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string solution(string X, string Y) {
    vector<int> cnts(10, 0);
    for(int i=0; i<X.length(); i++)
        cnts[X[i]-'0']++;
    
    string answer = "";
    for(int i=0; i<Y.length(); i++){
        if(cnts[Y[i]-'0']){
            answer += Y[i];
            cnts[Y[i]-'0']--;
        }
    }
    
    sort(answer.begin(), answer.end(), greater<char>());
    
    if(answer == "") return "-1";
    else if(answer[0] == '0') return "0";
    else return answer;
}
